sphinx/environment/__init__.py

```diff
@@ -240,12 +242,20 @@ def _update_config(self, config):
             for item in config.filter('env'):
                 if self.config[item.name] != item.value:
                     self.config_status = CONFIG_CHANGED
+                    self.config_status_extra = __(' (%r)') % (item.name,)
                     break
 
             # this value is not covered by the above loop because it is handled
             # specially by the config class
             if self.config.extensions != config.extensions:
                 self.config_status = CONFIG_EXTENSIONS_CHANGED
+                diff = sorted(
```

Comment:
mypy warns for this variable.
```
sphinx/environment/__init__.py: note: In member "_update_config" of class "BuildEnvironment":
sphinx/environment/__init__.py:255:21: error: Incompatible types in assignment (expression has type "str", variable has type "List[str]")
sphinx/environment/__init__.py:257:21: error: Incompatible types in assignment (expression has type "str", variable has type "List[str]")
```

How about renaming this list to `diffs` or `extensions`?